src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/AlignmentInterval.java

```diff
@@ -129,11 +171,54 @@ public AlignmentInterval(final String samSAtagString) {
         this.hasUndergoneOverlapRemoval = false;
     }
 
-    private static int parseInt(final String str) {
+    /**
+     * Returns the SA tag string representation for this interval.
+     * @return never {@code null}.
+     */
+    public String toSATagString() {
+        return appendSATagString(new StringBuilder(100)).toString();
+    }
+
+    @Override
+    public String toString() {
+        return toSATagString();
+    }
+
+    /**
+     * Appends the SA string representation of this interval into a builder.
+     * @param builder where to append to.
+     * @return reference to the input builder.
+     * @throws IllegalArgumentException if the builder is {@code null}.
+     */
+    public StringBuilder appendSATagString(final StringBuilder builder) {
+        Utils.nonNull(builder);
+        builder.append(referenceSpan.getContig()).append(SA_TAG_FIELD_SEPARATOR)
+               .append(referenceSpan.getStart()).append(SA_TAG_FIELD_SEPARATOR)
+               .append(forwardStrand ? SVFastqUtils.Strand.POSITIVE.toString() : SVFastqUtils.Strand.NEGATIVE.toString()).append(SA_TAG_FIELD_SEPARATOR)
+               .append(forwardStrand ? cigarAlong5to3DirectionOfContig.toString() : CigarUtils.invertCigar(cigarAlong5to3DirectionOfContig).toString()).append(SA_TAG_FIELD_SEPARATOR)
+               .append(mapQual);
+        if (mismatches != NO_NM || alnScore != NO_AS) {
+            builder.append(SA_TAG_FIELD_SEPARATOR);
+            if (mismatches == NO_NM) {
+                builder.append(NO_VALUE_STR).append(SA_TAG_FIELD_SEPARATOR).append(alnScore);
+            } else if (alnScore == NO_AS) {
+                builder.append(mismatches);
+            } else {
+                builder.append(mismatches).append(SA_TAG_FIELD_SEPARATOR).append(alnScore);
+            }
+        }
+        return builder;
+    }
+
+    private static int parseZeroOrPositiveInt(final String str, final int defaultValue, final String errorMessage) {
         try {
-            return Integer.parseInt(str);
-        } catch (NumberFormatException ex) {
-            throw new IllegalArgumentException("not a valid integer in: " + str, ex);
+            if (str.equals(NO_VALUE_STR)) {
+                return defaultValue;
+            } else {
+                return ParamUtils.isPositiveOrZero(Integer.parseInt(str), errorMessage + ": " + str);
```

Comment:
It would be clearer if the try...catch was around just this line rather than the whole method body.

Reply:
Ok.